src/jdk.charsets/share/classes/sun/nio/cs/ext/EUC_JP.java.template

```diff
@@ -113,14 +113,10 @@ public class EUC_JP
             byte[] sa = src.array();
             int sp = src.arrayOffset() + src.position();
             int sl = src.arrayOffset() + src.limit();
-            assert (sp <= sl);
```

Comment:
I see these are removed from encode/decodeArrayLoop(s). Any reason behind those?

Reply:
My IDE was marking the next line as redundant since `sp <= sl` will always be true. `buffer.position() <= buffer.limit()` is an invariant that I don't think we need to assert against here, and I think it's just been copy-pasted around mindlessly.

Reply:
Thanks, got it. I was wondering any possibility of sp > sl case.